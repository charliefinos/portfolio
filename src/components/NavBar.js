import React from 'react'
import { Link, Route } from 'react-router-dom'

const NavBar = () => {
    return (
        <div className="navbar">
            <nav className="nav">

                <ul className="nav-items">
                    <li className="nav-item" >

                        <Link to="/" data-testid="link-1">
                            Home
                        </Link>

                    </li>
                    <li className="nav-item">

                        <Link to="/about" >
                            About
                        </Link>

                    </li>
                    <li className="nav-item">

                        <Link to="/portfolio" >
                            Portfolio
                        </Link>

                    </li>
                    <li className="nav-item">

                        <Link to="/contact" >
                            Contact
                        </Link>

                    </li>
                </ul>
                <footer className="footer">
                    <p>
                        @2021 Carlos Finos
                    </p>
                </footer>
            </nav>
        </div >
    )
}

export default NavBar
