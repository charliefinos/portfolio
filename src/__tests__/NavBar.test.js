import { render, screen, cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import NavBar from '../components/NavBar';

afterEach(() => {
    cleanup();
})

test('should render NavBar component', () => {
    render(
        <Router>
            <NavBar exact={true} />
        </Router>
    );
    const linkElement = screen.getByTestId('link-1');
    expect(linkElement).toBeInTheDocument();
    expect(linkElement).toHaveTextContent('Home');
})

test('matches snapshot', () => {
    const tree = renderer.create(
        <Router>
            <NavBar exact={true} />
        </Router>).toJSON();

    expect(tree).toMatchSnapshot();
})