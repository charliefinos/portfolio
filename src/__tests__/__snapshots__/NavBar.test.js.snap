// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="navbar"
>
  <nav
    className="nav"
  >
    <ul
      className="nav-items"
    >
      <li
        className="nav-item"
      >
        <a
          data-testid="link-1"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          href="/portfolio"
          onClick={[Function]}
        >
          Portfolio
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          href="/contact"
          onClick={[Function]}
        >
          Contact
        </a>
      </li>
    </ul>
    <footer
      className="footer"
    >
      <p>
        @2021 Carlos Finos
      </p>
    </footer>
  </nav>
</div>
`;
